TempleEight/docs/arch/system.md

```diff
@@ -3,3 +3,189 @@ id: system
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+
+AnotherService: service {
+    baz: int;
+}
+```
+
+This results in the following changes to our architecture:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-two-services.png')} />
+
+We can see that all of the existing infrastructure that supported `ExampleService` has been replicated for `AnotherService`, but everything else remains the same.
+We can extend this further by making a Foreign Key reference between services (See the [Foreign Keys](../guide/foreign-keys) guide for full details). 
+
+```templefile {10}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  foreignKey: AnotherService;
+}
+
+AnotherService: service {
+    baz: int;
+}
+```
+
+This has the effect of generating inter-service communication logic in the service-level application code.
+On the project level, nothing changes, except the services use the hostname resolution of the orchestration platform to make HTTP requests to each other.
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-foreign-key.png')} />
+
+### Structs
+
+Adding a struct to a service in your Templefile represents two data items that are strongly coupled (see the [Struct](../guide/structs) guide for details).
+This is implemented in Temple generated projects by extending the parent's application executable to include the additional endpoints required to process struct requests.
+The struct's data is stored in a separate table of the service's database. 
+This means that when adding structs to your Templefile, the overall system-level architecture doesn't change.
+
+### Auth
+
+As per the [Authentication](../guide/authentication) guide, 
+authentication can be added to your system by augmenting your Templefile to include `#auth` metadata on services, and a `#authMethod` annotation on the project block.
+For example:
+
+```templefile {5,10}
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+  #authMethod(email);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+  #auth;
+}
+```
+
+These changes result in an additional `Auth` service being generated automatically that stores and verifies user credentials, and issues authentication tokens to users for use with other services.
+This is implemented by configuring Kong to check that any incoming requests have a valid [JWT](https://jwt.io/), except those destined for the auth service. 
+Any requests that are not correctly authenticated do not pass Kong, so the internal microservices can be sure that all traffic has been correctly authenticated.
+
+Visually, the changes to our architecture look like:
+
+<img alt="System Architecture Kube" src={useBaseUrl('img/system-architecture-auth.png')} />
```

Comment:
not a giant fan of the the star, seems a bit out of place, but if u want it I'm cool with it

please make the text not clip the edges though (of the star or the `C` of the `Docker Container` label)